src/app/_services/image-group.service.ts

```diff
@@ -72,71 +76,74 @@ export class ImageGroupService {
   }
 
   public triggerPPTExport(): void {
-    this.igDownloadTrigger.emit('ppt');
+    this.igDownloadTrigger.emit(SupportedExportTypes.PPTX);
   }
 
   public triggerZIPExport(): void {
-    this.igDownloadTrigger.emit('zip');
+    debugger;
+    this.igDownloadTrigger.emit(SupportedExportTypes.ZIP);
   }
 
   public triggerGoogleSlides(): void {
     this.igDownloadTrigger.emit('GoogleSlides');
   }
 
-  /** Gets the link at which the resource can be downloaded. Will be set to the "accept" button's download property */
-  public getDownloadLink(group: ImageGroup, zip ?: boolean): Promise<any> {
-    let header = new HttpHeaders().set('content-type', 'application/x-www-form-urlencoded')
-    let options = { headers: header, withCredentials: true }
-    let useLegacyMetadata: boolean = true
-    let url = this._auth.getHostname() + '/api/group/export'
-    let format: string
-    let data: any
-
-    if (!zip) {
-      format = 'pptx'
-    } else {
-      format = 'zip'
-    }
+  /**
+   * Given the image group determine the ids that are to be download given the user's access.
+   */
+  public getImageIdsToDownload(group: ImageGroup) {
```

Comment:
Needed to extract this logic out into it's own function because we need to remember which images are actually being downloaded when we are ready to log them.